verl/workers/critic/dp_critic.py

```diff
@@ -176,12 +176,15 @@ def compute_values(self, data: DataProto) -> torch.Tensor:
 
     def update_critic(self, data: DataProto):
         # make sure we are in training mode
+        multi_turn = data.meta_info.get("multi_turn", False)
         self.critic_module.train()
         metrics = {}
 
         select_keys = ["input_ids", "responses", "attention_mask", "position_ids", "values", "returns"]
+        if multi_turn:
+            select_keys.append("loss_mask")
         batch = data.select(batch_keys=select_keys).batch
-        has_multi_modal_inputs = "multi_modal_inputs" in data.non_tensor_batch.keys()
+        has_multi_modal_inputs = "multi_modal_inputs" in data.non_tensor_batch
 
```

Comment:
i think ruff is too strict (and later ignored) and you need to put back `.keys()` to be compatible with tensor-dict

Reply:
same for other .keys() calls